File: rBike.Model/Review.cs
using System;

namespace rBike.Model
{
    public class Review
    {
        public int BikeReviewId { get; set; }

        public int BikeId { get; set; }

        public int UserId { get; set; }

        public int Rating { get; set; }

        public DateTime Date { get; set; }

        public virtual Bike Bike { get; set; } = null!;

        public virtual User User { get; set; } = null!;

        public string? Username { get; set; }

        public string? BikeName { get; set; }
    }
}

File: UI/rbike_mobile/lib/models/review.dart
import 'package:json_annotation/json_annotation.dart';

part 'review.g.dart';

@JsonSerializable()
class Review {
  int? bikeReviewId;
  int? bikeId;
  int? userId;
  int? rating;
  @JsonKey(name: 'date')
  DateTime? datum;
  String? bikeName;
  String? username;

  Review();

  factory Review.fromJson(Map<String, dynamic> json) => _$ReviewFromJson(json);

  Map<String, dynamic> toJson() => _$ReviewToJson(this);
} 